test/release/examples/primers/modules.chpl

```diff
@@ -0,0 +1,390 @@
+    */
+    use modToUse;
+
+    /*
+      Use statements can be inserted at the global scope, as well as any
+      arbitrary inner scope defined.
+    */
+
+    /* A use statement brings all of the visible symbols of a module into the
+       scope which defines the use statement.  These symbols may then be
+       accessed without the module name prefix.
+    */
+    var bazBarFoo = baz(bar, foo); // should be 28
+    writeln(bazBarFoo);
+  }
+
+
+
+  /* Once the scope defining a use statement is left, however, the symbols must
+     be accessed with a prefix.  The following line would cause an error,
+     because the previous use of the module 'modToUse' occurred within the curly
+     braces above:
+  */
+  // var twiceFoo = 2 * foo;
+
+
+
+  {
+    var bazBarFoo = baz(bar, foo); // should be 28
+
+    /* Use statements apply to the entire scope in which they are defined.  Even
+       if the use statement occurs after code which would access the symbols
+       without prefix, these accesses are still valid.
+    */
+    use modToUse;
+
+    writeln(bazBarFoo);
+  }
+
+
+  {
+    /* The symbols provided by a use statement are considered at an outer scope
+       to the symbols defined at the same level as the use itself.  Thus, if
+       another bar were defined here, and an access was attempted, the compiler
+       would find the bar at this scope, rather than modToUse.bar.
+    */
+    var bar = 4.0;
+
+    use modToUse;
+
+    writeln(bar); // Will output 4.0, not 2
+  }
+
+
+  var bar = false;
+  {
+    /* The symbols provided by a use statement are considered at an inner scope
+       to the symbols defined outside of the scope where the use statement
+       applies.  Thus, if another bar were defined outside of these curly
+       braces, and an access was attempted, the compiler would find the bar from
+       modToUse, rather than the outer bar.
+    */
+
+    use modToUse;
+    writeln(bar); // Will output 2, not false
+  }
+
+
+  {
+    /* Multiple modules may be used in the same use statement */
+    use modToUse, AnotherModule, ThirdModule;
+
+    if (a || b < 0) {
+      writeln(foo);
+    } else {
+      writeln(bar);
+    } // Will output foo (12)
+  }
+
+
+  {
+    /* And a scope may contain multiple use statements */
+    use modToUse;
+    use AnotherModule;
+
+    writeln(a && foo > 15); // outputs false
+  }
+
+
+  {
+    /* In either case, the modules used in this way are considered at the same
+       scope.  This means that if two modules each define a symbol with the same
+       name, and both modules are used at the same scope, attempts to access a
+       symbol by that name will result in a naming conflict:
+    */
+    use modToUse, Conflict;
+
+    writeln(foo); // Outputs 12
+    /* The following line would fail because both modToUse and Conflict define a
+       bar:
+    */
+    // writeln(bar);
+    writeln(other); // Outputs 5.0 + 3.0i
+  }
+
+  writeln();
+  writeln("Limiting a use");
```

Comment:
The intention was to break up the output so I could more easily tell if a section was behaving as I expected it to.  I was trying to avoid annotating all the output lines to optimize time writing the output versus time verifying the output is correct.